src/mappings/markets.ts

```diff
@@ -49,125 +81,316 @@ function getTokenPrice(
+
+    market.underlyingDecimals = erc20_decimals(
+      market.underlyingAddress.toHex(),
+      underlyingContract,
+    )
+    log.info('MARKETS::createMarket->3 {}', [market.underlyingAddress.toHex()])
+
+    market.underlyingName = erc20_symbol(
+      market.underlyingAddress.toHex(),
+      underlyingContract,
+    )
+    market.underlyingSymbol = erc20_symbol(
+      market.underlyingAddress.toHex(),
+      underlyingContract,
+    )
+
+    if (marketAddress == cUSDC_ADDRESS) {
       market.underlyingPriceUSD = BigDecimal.fromString('1')
     }
   }
 
-  market.borrowRate = zeroBD
-  market.cash = zeroBD
-  market.collateralFactor = zeroBD
-  market.exchangeRate = zeroBD
+  market.borrowRate = ZERO_BD
+  market.borrowAPY = ZERO_BD
+  market.borrowDistributionAPY = ZERO_BD
+  market.cash = ZERO_BD
+  market.collateralFactor = ZERO_BD
+  market.exchangeRate = ZERO_BD
   market.interestRateModelAddress = Address.fromString(
     '0x0000000000000000000000000000000000000000',
   )
-  market.name = contract.name()
+
+  market.name = cToken_symbol(marketAddress, contract)
   market.numberOfBorrowers = 0
   market.numberOfSuppliers = 0
-  market.reserves = zeroBD
-  market.supplyRate = zeroBD
-  market.symbol = contract.symbol()
-  market.totalBorrows = zeroBD
-  market.totalSupply = zeroBD
-  market.underlyingPrice = zeroBD
+  market.reserves = ZERO_BD
+  market.supplyRate = ZERO_BD
+  market.supplyAPY = ZERO_BD
+  market.supplyDistributionAPY = ZERO_BD
+  market.symbol = cToken_symbol(marketAddress, contract)
+  market.totalBorrows = ZERO_BD
+  market.totalSupply = ZERO_BD
+  market.underlyingPrice = ZERO_BD
 
   market.accrualBlockNumber = 0
   market.blockTimestamp = 0
-  market.borrowIndex = zeroBD
+  market.borrowIndex = ZERO_BD
   market.reserveFactor = BigInt.fromI32(0)
-  market.underlyingPriceUSD = zeroBD
+  market.underlyingPriceUSD = ZERO_BD
 
   return market
 }
 
+function erc20_decimals(address: string, contract: ERC20): i32 {
+  let staticDefinition = TokenDefinition.fromAddress(Address.fromString(address))
+  if (staticDefinition != null) {
+    return (staticDefinition as TokenDefinition).decimals
+  }
+
+  // HACKY - handling of runtine errors
+  let decimals: i32 = 0
+  let result = contract.try_decimals()
+  if (!result.reverted) {
+    decimals = result.value
+  } else {
+    log.info('CUSTOM' + address.toString(), [])
+  }
+
+  return decimals
+}
+
+function erc20_symbol(address: string, contract: ERC20): string {
+  let staticDefinition = TokenDefinition.fromAddress(Address.fromString(address))
+  if (staticDefinition != null) {
+    return (staticDefinition as TokenDefinition).symbol
+  }
+
+  // HACKY - handling of runtine errors
+  let symbol = 'N/A'
+  let result = contract.try_symbol()
+  if (!result.reverted) {
+    symbol = result.value
+  } else {
+    log.info('CUSTOM' + address.toString(), [])
+  }
+
+  return symbol
+}
+
+function cToken_name(marketAddress: string, contract: CToken): string {
```

Comment:
let remove this hardcoding for name, symbol, decimals etc

Reply:
resolved